### PR TITLE
fix(platform): Bad links when selecting tags on GroupEventDetails to narrow search

### PR DESCRIPTION
**Pull request by leedongwei**
Fix regression introduced in #14524

On `GroupEventDetails`, the users expect to narrow down their search when clicking on the `Pill`. However, if the key-value of the `Pill` was project/environment, it would unexpectedly change the state of the GlobalSelectionHeader due to the implementation of the function `generateQueryWithTag`.

On the other hand, clicking the facets/heatmap in Discover2 uses this wrong implementation of `generateQueryWithTag` (but it resulted in the correct behaviour for Discover2). To allow for this, a separate function for this behaviour was created in `app/views/eventsV2/utils.tsx`.

There were discussions on wrapping tag's keys in this format `tag[myKey]` but Discover2 is not ready for it right now.

Refs SEN-1069
Refs SEN-1124

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/queryString.tsx [hunk 1]
```diff
@@ -16,41 +17,127 @@ export function addQueryParamsToExistingUrl(
...
+    ? query
+    : Array.isArray(query)
+    ? query.reduce((acc, q) => `${acc} ${q}`, '')
+    : '';
+
+  // 1) Start with the QueryValue: "key1:value1 key2:value2"
+  // 2) Split it into their pairs: ["key1:value1", "key2:value2"]
+  // 3) Split it into key-values:  { key1: value1, key2: value2 }
+  return currQuery.split(' ').reduce((acc, keyValue) => {
+    const pair = keyValue.split(':');
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/queryString.tsx]: Quoted values could also contain `:`. For example `mytag:"namespace:value"`

File: src/sentry/static/sentry/app/utils/queryString.tsx [hunk 2]
```diff
@@ -16,41 +17,127 @@ export function addQueryParamsToExistingUrl(
...
+  const currQuery: string = isString(query)
+    ? query
+    : Array.isArray(query)
+    ? query.reduce((acc, q) => `${acc} ${q}`, '')
+    : '';
+
+  // 1) Start with the QueryValue: "key1:value1 key2:value2"
+  // 2) Split it into their pairs: ["key1:value1", "key2:value2"]
+  // 3) Split it into key-values:  { key1: value1, key2: value2 }
+  return currQuery.split(' ').reduce((acc, keyValue) => {
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/queryString.tsx]: Quoted values can contain spaces. eg `message:"ValueError foo"`

File: src/sentry/static/sentry/app/utils/queryString.tsx [hunk 3]
```diff
@@ -16,41 +17,194 @@ export function addQueryParamsToExistingUrl(
...
+
+const regexValue = /^[a-zA-Z0-9_\.-]+$/;
+const regexValueWithSpace = /^[a-zA-Z0-9_\.-\s]+$/;
+const regexValueWithSpaceQuoted = /^"[a-zA-Z0-9_\.-\s]+"$/;
+const regexValueWithNamespaceSpace = /^[a-zA-Z0-9_\.-\s]+:(\\\")[a-zA-Z0-9_\.-\s]+(\\\")$/;
+const regexValueWithNamespaceSpaceQuoted = /^"[a-zA-Z0-9_\.-\s]+:(\\\")[a-zA-Z0-9_\.-\s]+(\\\")"$/;
+/**
+ * This will add TODO(leedongwei)
+ */
+export function _encodeValue(value: string): string {
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/queryString.tsx]: Why is a `_` prefixed function being exported? Should the `_` be removed?

File: src/sentry/conf/server.py [hunk 4]
```diff
@@ -804,17 +804,17 @@ def create_partitioned_queues(name):
...
+    "organizations:discover": True,
     # Enable attaching arbitrary files to events.
     "organizations:event-attachments": False,
     # Allow organizations to configure built-in symbol sources.
     "organizations:symbol-sources": True,
     # Allow organizations to configure custom external symbol sources.
     "organizations:custom-symbol-sources": True,
     # Enable the events stream interface.
-    "organizations:events": False,
+    "organizations:events": True,
```
Comment (markstory) [inline src/sentry/conf/server.py:815]: Not sure you want to do this.

File: src/sentry/static/sentry/app/utils/queryString.tsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/utils/queryString.tsx:121]: Would be nice to have a query tokenizer on the frontend

**Comment by BYK** [comment]
Closing due to staleness and conflicts.

**Pull request closed without merging**